### ISSUE TITLE
Unable to reproduce the OpenCLIP ViT-B/32 `+VL-SAE` results in Table 1

### ISSUE THREAD

**Issue by shl-shawn** [ssfgunner/VL-SAE#7]
Hi authors,

Thank you for releasing the paper, implementation, and pretrained weights. We are trying to reproduce the OpenCLIP ViT-B/32 zero-shot classification results in Table 1 before building on VL-SAE. The raw OpenCLIP results reproduce closely, but we have not been able to reproduce the reported `+VL-SAE` improvements using the released checkpoint.

## Setup

- OpenCLIP model: `ViT-B-32`
- OpenCLIP pretrained weights: `laion2b_s34b_b79k`
- Released VL-SAE checkpoint: `openclip_ViT-B-32_VL_SAE_256_8_best.pth`
- Checkpoint SHA-256: `41dff216ab19a055db30d71a0d094da9bceabc0a2bc317cb64dc4ba8b80afeb1`
- VL-SAE hidden dimension: 4096 (`hidden_ratio=8`)
- Top-K: 256
- Evaluation: CLIP Benchmark class names, prompt templates, and test datasets
- `open_clip_torch==2.29.0`, `clip-benchmark==1.6.2`, `torch==2.0.1`

We used the dataset-specific alpha values reported in Table A6.

## Results

Top-1 accuracy:

| Dataset | Paper: raw | Ours: raw | Paper: +VL-SAE | Ours: Eq. 10, hidden activations | Ours: decoder reconstructions |
| --- | ---: | ---: | ---: | ---: | ---: |
| CIFAR-10 | 93.6 | 93.56 | 93.8 | 93.79 | 93.78 |
| DTD | 55.9 | 56.01 | 56.9 | 56.38 | 56.17 |
| EuroSAT | 47.2 | 47.15 | 51.0 | 46.03 | 47.27 |
| MNIST | 69.8 | 69.96 | 71.8 | 62.51 | 58.79 |

The close agreement of the raw results suggests that the base model, datasets, preprocessing, class names, and prompt templates are consistent with Table 1.

## Implementations we tested

First, we followed Equation 10 literally:

```text
score = cosine(x_v, x_l) + alpha * cosine(h_v, h_l)
```

Here, `h_v` and `h_l` are produced by the released `VL_SAE.encode` distance encoder followed by Top-256 sparsification. This gives the “Eq. 10, hidden activations” column above.

We also noticed that `OPENCLIP_VLSAE.encode_image_enhance` and `encode_text_enhance` return outputs `[0]` and `[1]` from `VL_SAE.forward`, which appear to be the modality-specific decoder reconstructions rather than hidden concept activations. We therefore also tested:

```text
score = cosine(x_v, x_l) + alpha * cosine(recon_v, recon_l)
```

This gives the final column, but it also does not reproduce Table 1.

One possible ambiguity is whether textual prompts should be aggregated before or after VL-SAE encoding. However, MNIST uses only one prompt template in our evaluation, so this ambiguity cannot explain the large MNIST discrepancy.

## Questions

Could you please clarify the exact Table 1 evaluation protocol?

1. Does Equation 10 use the sparse hidden activations `h_v, h_l`, decoder reconstructions, or another representation?
2. Is the score exactly the additive cosine score in Equation 10, or is it combined at the probability/logit level or other else?
3. How are multiple text prompt templates aggregated relative to VL-SAE encoding and normalization?
4. Was the released checkpoint with the SHA-256 above the checkpoint used to produce the ViT-B/32 row in Table 1?
5. Were the alpha values in Table A6 applied directly to the two cosine similarities?

If possible, could you release the zero-shot classification script/configuration used for Table 1? Alternatively, even a small code fragment showing how the image features, prompt features, VL-SAE features, and final logits were computed would help us identify the discrepancy.

We would be happy to share our reproduction code or run any diagnostic you recommend. Thank you for your time and help.
